cli/tests/pcluster/validators/test_cluster_validators.py

```diff
@@ -1137,3 +1138,25 @@ def test_mixed_security_group_overwrite_validator(head_node_security_groups, que
     )
     expected_message = "make sure.*cluster nodes are reachable" if expect_warning else None
     assert_failure_messages(actual_failures, expected_message)
+
+
+@pytest.mark.parametrize(
+    "root_volume_size, expected_message, ami_size",
+    [
+        (65, None, 50),
+        (25, None, 8),
+        (
+            25,
+            "Root volume size 25 GB must be equal or greater than .* 50 GB.",
+            50,
+        ),
+    ],
+)
+def test_root_volume_size_validator(mocker, root_volume_size, expected_message, ami_size):
+    mock_aws_api(mocker)
+    mocker.patch(
+        "pcluster.aws.ec2.Ec2Client.describe_image",
+        return_value=ImageInfo({"BlockDeviceMappings": [{"Ebs": {"VolumeSize": ami_size}}]}),
+    )
+    actual_failures = RootVolumeSizeValidator().execute(root_volume_size, "ami-123456789a8a37250")
+    assert_failure_messages(actual_failures, expected_message)
```

Comment:
We are testing the failure messages, but we are not testing the failure type (ERROR vs WARNING).
I know that it the majority of our tests we do not cover it, but I think it's good to.
Let's consider this as an improvement out of scope for this PR, but let's be sure to track t.